src/tablecloth/api/join_separate.clj

```diff
@@ -62,11 +62,20 @@
        (process-join-columns ds target-column join-function col-names drop-columns?)))))
 
 ;;
-
 (defn- infer-target-columns
   [col res]
-  (let [colname (col/column-name col)]
-    (map #(str colname "-" %) (range (count (first res))))))
+  (let [colname (col/column-name col)
+        update-f (fn [colname value]
+                   (let [with-prefix (str (->str colname) "-" value)]
+                     (if (keyword? colname) (keyword with-prefix) with-prefix)))]
+    (map #(update-f colname %) (range (count (first res))))))
+
+(defn- prefix [prefix-name value]
```

Comment:
This is another function with the same name (it's defined also later).

I propose to move it above `infer-target-columns` and delete redundand copies.